shell/platform/fuchsia/dart_runner/embedder/BUILD.gn

```diff
@@ -14,7 +14,7 @@ template("dart_shim_kernel") {
     main_dart = rebase_path(invoker.main_dart)
 
     deps = [
-      "../kernel:kernel_platform_files($host_toolchain)",
+      "../kernel:kernel_platform_files",
```

Comment:
@chinmaygarde , @jason-simmons maybe you guys know this, but when i specify the `($host_toolchain)` here gn is unable to resolve the dependency of the input `platform_dill` between this and the `kernel_platform_files` target.

My theory is that maybe data deps can only exist between targets of same toolchain.

Reply:
I was able to look into `prebuilt_dart_action` and it seems to be relying on the `host_toolchain` by default and the redundant dependency here seems to be causing the problem. This PR should be good.